converter/src/main/java/gov/cms/qpp/conversion/decode/QrdaDecoder.java

```diff
@@ -0,0 +1,98 @@
+package gov.cms.qpp.conversion.decode;
+
+import com.google.common.base.Strings;
+import org.jdom2.Attribute;
+import org.jdom2.Element;
+import org.jdom2.Namespace;
+import org.jdom2.filter.Filter;
+import org.jdom2.filter.Filters;
+import org.jdom2.xpath.XPathExpression;
+import org.jdom2.xpath.XPathFactory;
+
+import gov.cms.qpp.conversion.Context;
+import gov.cms.qpp.conversion.correlation.PathCorrelator;
+import gov.cms.qpp.conversion.model.Decoder;
+import gov.cms.qpp.conversion.model.Node;
+import gov.cms.qpp.conversion.model.validation.SupplementalData;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Consumer;
+
+import static gov.cms.qpp.conversion.decode.SupplementalDataEthnicityDecoder.SUPPLEMENTAL_DATA_CODE;
+import static gov.cms.qpp.conversion.decode.SupplementalDataEthnicityDecoder.SUPPLEMENTAL_DATA_KEY;
+import static gov.cms.qpp.conversion.decode.SupplementalDataPayerDecoder.SUPPLEMENTAL_DATA_PAYER_CODE;
+
+public abstract class QrdaDecoder {
+
+	protected final Context context;
+	private Namespace defaultNs;
+	protected Namespace xpathNs;
+
+	public QrdaDecoder(final Context context) {
+		this.context = context;
+	}
+
+	protected abstract DecodeResult decode(Element element, Node thisNode);
+
+	/**
+	 * Sets xml namespace
+	 */
+	public void setNamespace(Namespace defaultNs) {
+		this.defaultNs = defaultNs;
+		String defaultNsUri = defaultNs.getURI();
+		xpathNs = Strings.isNullOrEmpty(defaultNsUri) ? Namespace.NO_NAMESPACE : Namespace.getNamespace("ns", defaultNsUri);
+	}
+
+	/**
+	 * Returns the xpath from the path-correlation.json meta data
+	 *
+	 * @param attribute Key to the correlation data
+	 * @return xpath expression as a string
+	 */
+	protected String getXpath(String attribute) {
+		String template = this.getClass().getAnnotation(Decoder.class).value().name();
+		return PathCorrelator.getXpath(template, attribute, defaultNs.getURI());
+	}
+
+	/**
+	 * Executes an Xpath for an element and executes the consumer
+	 *
+	 * @param element Element the xpath is executed against
+	 * @param expressionStr Xpath
+	 * @param consumer Consumer to execute if the xpath matches
+	 * @param filter Filter to apply for the xpath
+	 * @param selectOne Whether to execute for the first match or multiple matches
+	 */
+	@SuppressWarnings({ "rawtypes", "unchecked" })
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing warnings is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1309)